activerecord/lib/active_record/connection_adapters/postgresql/oid/hstore.rb

```diff
@@ -24,6 +24,8 @@ def deserialize(value)
           def serialize(value)
             if value.is_a?(::Hash)
               value.map { |k, v| "#{escape_hstore(k)}=>#{escape_hstore(v)}" }.join(", ")
+            elsif value.respond_to?(:to_unsafe_h)
```

Comment:
Because this is fundamentally different than attribute assignment elsewhere. It's assigning a value to a single known attribute. Similarly to how we don't require users to whitelist possible values for a string when assigning to a string column. Most importantly, this is matching the behavior of 4.2